host/crypto/openssl/key.c

```diff
@@ -41,6 +42,83 @@ void oe_private_key_init(
     impl->pkey = pkey;
 }
 
+oe_result_t oe_private_key_from_engine(
+    const char* engine_id,
+    const char* engine_load_path,
+    const char* key_id,
+    oe_private_key_t* key,
+    int key_type,
+    uint64_t magic)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    oe_private_key_t* impl = key;
+    BIO* bio = NULL;
+    EVP_PKEY* pkey = NULL;
+
+    /* Initialize the key output parameter */
+    if (impl)
+        oe_secure_zero_fill(impl, sizeof(oe_private_key_t));
+
+    /* Check parameters */
+    if (!engine_id || !engine_load_path || !key_id || !impl)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    /* if the engine is pre-installed we will get a null */
+    ENGINE* e = NULL;
+    if (!(e = ENGINE_by_id(engine_id)))
+    {
+        /* if we got a null we can still load dynaimc. */
+        ENGINE_load_dynamic();
```

Comment:
Consider forcing dynamic loading if `engine_load_path` is non-null. Dynamic loading as an OpenSSL feature is also used for creating a separate instance of the engine for the caller, and someone who's using this feature may not want to get the default registered instance when they've already explicitly specified the load path (and hence opt-in to dynamic loading)

Reply:
done